### PR TITLE
support both CHAP methods

### PR DESCRIPTION
**Pull request by benwwchen**
According to #11, not all authorization servers have upgraded to the new CHAP method.

This commit lets you choose between old and new methods. See [README.md](https://github.com/benwwchen/sysuh3c/blob/openwrt/README.md).

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -213,6 +217,13 @@ int main(int argc, char **argv) {
             case 'i':
                 strcpy(iface, optarg);
                 break;
+            case 'm':
+                if (strlen(optarg) > 1) {
```
Comment (zonyitoo) [inline src/main.c]: what if `strlen(optarg) == 0`?

File: src/main.c [hunk 2]
```diff
@@ -250,7 +261,12 @@ int main(int argc, char **argv) {
         free(expr);
     }
 
-    if (eapauth_init(&eapauth, iface) != 0)
+    if (method != '0' && method != '1') {
```
Comment (zonyitoo) [inline src/main.c]: '0' and '1' now become magic code, could you define an `enum` or something?

File: src/eapauth.c [hunk 3]
```diff
@@ -269,22 +273,32 @@ int send_response_h3c(const eapauth_t *user, uint8_t packet_id) {
...
     uint8_t chapbuf[128] = {0}; // id + password + md5data
-    size_t chapbuflen, passwordlen;
+    size_t chapbuflen, passwordlen, i;
     uint8_t packetbuf[128] = {0};
 
     if (user == NULL || md5data == NULL) return EAPAUTH_ERR;
 
-    passwordlen = strlen(user->password);
-    chapbuflen = 1 + passwordlen + 16;
+    switch(md5_method) {
```
Comment (zonyitoo) [inline src/eapauth.c]: it is not robust to put `md5_method` in global scope

File: src/eapauth.c [hunk 4]
```diff
@@ -59,7 +61,7 @@ void display_promote_func(int priority, const char *format, ...) {
 static void (*status_notify)(int) = status_notify_func;
 static void (*display_promote)(int, const char *, ...) = display_promote_func;
 
-int eapauth_init(eapauth_t *user, const char *iface) {
+int eapauth_init(eapauth_t *user, const char *iface, char method) {
```
Comment (zonyitoo) [inline src/eapauth.c]: you can put `method` into `eapauth_t` struct

Comment (zonyitoo) [reply src/eapauth.c]: and I also suggest to define `method` as an `enum` instead of a `char`

File: README.md [hunk 5]
```diff
@@ -34,6 +34,9 @@ endef
 -u --user        user account
 -p --password    password
 -i --iface       network interface (default is eth0)
+-m --method      EAP-MD5 CHAP method (default 0)
```
Comment (zonyitoo) [inline README.md]: It would be more human friendly to pass parameter as string, for instance, `-m xor`

**Review by zonyitoo — CHANGES_REQUESTED** [review]

**Comment by benwwchen** [comment]
@zonyitoo Please check the new commit. Thanks.

**Review by zonyitoo — APPROVED** [review]

**Pull request merged**